component/otelcol/exporter/prometheus/internal/convert/cache.go

```diff
@@ -101,6 +102,19 @@ func (series *memorySeries) WriteTo(app storage.Appender, ts time.Time) error {
 	if newID != series.id {
 		series.id = newID
 	}
+
+	return nil
+}
+
+func (series *memorySeries) WriteExemplarsTo(app storage.Appender, e exemplar.Exemplar) error {
+	series.Lock()
+	defer series.Unlock()
+
+	_, err := app.AppendExemplar(series.id, series.labels, e)
+	if err != nil {
+		return err
+	}
+
```

Comment:
Here I would actually suggest a different change:
```
	if _, err := app.AppendExemplar(series.id, series.labels, e); err != nil {
		return err
	}
	return nil
```
Although this is a bit more verbose, I found returning "err" misleading because in most areas when "err" is returned you expect that it is != nil